Blocks/Persistence/file.h

```diff
@@ -430,7 +442,10 @@ class FilePersister {
     }
   }
 
-  std::chrono::microseconds CurrentHead() const noexcept { return file_persister_impl_->end.load().head; }
+  template <current::locks::MutexLockStatus>
+  std::chrono::microseconds CurrentHead() const noexcept {
```

Comment:
You mean inconsistency with the `LastPublishedIndexAndTimestamp`? I thought we agreed on adding the template parameter only to those methods, which need it because they are used from inside the `StreamImpl`.